UserBundle/Validator/Constraints/ComplexUserPasswordValidator.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace OpenOrchestra\UserBundle\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Class ComplexUserPasswordValidator
+ */
+class ComplexUserPasswordValidator extends ConstraintValidator
+{
+    /**
+     * @param string     $value
+     * @param Constraint $constraint
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (strlen($value) < 8
```

Comment:
It could be done with only one regex : ^(?=.*[A-Z])(?=.*[a-z])(?=.*[$@&_#\*\-\+]).{8,}$